accounts/abi/bind/base.go

```diff
@@ -340,6 +368,58 @@ func (c *BoundContract) UnpackLog(out interface{}, event string, log types.Log)
 	return parseTopics(out, indexed, log.Topics[1:])
 }
 
+// preparePrivateTransaction replace the payload of private transaction to the hash from Tessera/Constellation
+func (c *BoundContract) preparePrivateTransaction(tx *types.Transaction, payload []byte) *types.Transaction {
+	var privateTx *types.Transaction
+	if tx.To() == nil {
+		privateTx = types.NewContractCreation(tx.Nonce(), tx.Value(), tx.Gas(), tx.GasPrice(), payload)
+	} else {
+		privateTx = types.NewTransaction(tx.Nonce(), c.address, tx.Value(), tx.Gas(), tx.GasPrice(), payload)
+	}
+	privateTx.SetPrivate()
+	return privateTx
+}
+
+// sendPrivateTx send the private transaction to Tessera/Constellation
+func sendPrivateTx(tx *types.Transaction, privateFrom string, url string) ([]byte, error) {
+	if private.P == nil {
+		return nil, errors.New("transaction manager not set up")
+	}
+	data, err := storeRaw(tx.Data(), privateFrom, url)
+	if err != nil {
+		return nil, err
+	}
+	return data, nil
+}
+
+// storeraw API params
+type StoreRawArgs struct {
+	Payload string `json:"payload"`
+	From    string `json:"from"`
+}
+type StoreRawResp struct {
+	Key string `json:"key"`
+}
+
+// storeRaw calls the /storeraw API of tessera
+func storeRaw(data []byte, privateFrom string, url string) ([]byte, error) {
+	reqBodyJSON := &StoreRawArgs{
+		Payload: string(data),
+		From:    privateFrom}
+	reqBody, _ := json.Marshal(reqBodyJSON)
+	resp, err := http.Post(url+"/storeraw", "application/json", bytes.NewBuffer(reqBody))
```

Comment:
Private Transaction Manager (Tessera) third-party API server supports [multiple ways of communications](http://docs.goquorum.com/en/latest/Privacy/Tessera/Configuration/Configuration%20Overview/#server). If `ethclient` only support HTTP, need to be explicit and handling check explicitly

Reply:
Make `ethclient` function name explicit as `StoreRawHTTP`